tests/test_umath.py

```diff
@@ -1,25 +1,13 @@
-"""
-Tests of the code in uncertainties.umath.
-
-These tests can be run through the Nose testing framework.
-
-(c) 2010-2016 by Eric O. LEBIGOT (EOL).
-"""
-
-from __future__ import division
-from __future__ import absolute_import
-
-# Standard modules
 import sys
 import math
+from math import isnan, isinf
 
-# Local modules:
 from uncertainties import ufloat
 import uncertainties.core as uncert_core
 import uncertainties.umath_core as umath_core
 
-from . import test_uncertainties
-
+from uncertainties.testing import compare_derivatives, numbers_close
+from test_uncertainties import power_special_cases, power_all_cases, power_wrt_ref
```

Comment:
It does not seem right to import from one `test_` file to another. These functions do seem too specialized for `uncertainties.testing` though. Could we use one of the options in [this StackOverflow question](https://stackoverflow.com/questions/33508060/create-and-import-helper-functions-in-tests-without-creating-packages-in-test-di). Maybe the `conftest.py` and fixture option, or the `pythonpath` and helpers file option?

Reply:
Adding `testpaths = ["tests"]` is not required for tests to work right? It's just a convenience that makes `pytest` discover tests more quickly? Is my understanding correct?

In the past I've put helper function needed for testing in an `__init__.py` file in the `tests` folder. These objects can then be imported within the test scripts. We could also have a `tests/helper.py` module to accomplish the same result.

Reply:
Yea, importing across test files does feel a bit odd. The helpers.py solution is nice and simple.

Reply:
> Adding testpaths = ["tests"] is not required for tests to work right? It's just a convenience that makes pytest discover tests more quickly? Is my understanding correct?

Yes, that seems to be all it does functionally. My conjecture was that it might also help IDE's figure out that the project uses pytest as the test runner because otherwise there is no configuration indicating that pytest should be run, but I haven't tried to confirm that.

I like the move to helpers.py. We might want to add

```
testpaths = ["tests"]
addopts = [
    "--import-mode=importlib",
]
```

to pyproject.toml like the one StackOverflow answer suggests, but we could wait for that. Currently, pytest adds all the test directories to the Python path so you can do `import helpers` but it has been trying to move away from that to just importing the test files with `importlib` without modifying the Python path.

Reply:
thanks for explaining, wouldn't have understood why otherwise